platform-plugins.md

```diff
@@ -0,0 +1,277 @@
+(like `Clipboard` and `BatteryState`) and third-party platform SDKs (like
+`Google Analytics`).
+
+* TOC
+{:toc}
+
+## Options for calling platform-specific APIs
+
+Mobile devices offer many options to extend your app through platform-specific
+APIs. These are critical to the completeness of most mobile apps, and Flutter
+imposes no limits on your ability to them.
+
+Flutter offers the following ways to use platform-specific services:
+
+* **Use existing Platform Plugins:** For frequently used APIs and SDKs, use
+ existing Platform Plugins, with no need to write any platform-specific code.
+ [See details below](#use).
+
+* **Create a Platform Plugin:** Write your own platform-specific code using
+ Flutter's Platform Channels. This can be used for private APIs, or for cases
+ where a suitable Platform Plugin is not yet available. [See details below](#create).
+
+* **Publish a Platform Plugin:** Publish your own custom platform-specific as a
+ plugin. This enables others to use it by simply declaring a dependency on the
+ plugin. [See details below](#publish).
+
+## Use existing Platform Plugins {#use}
+
+### Core Flutter Platform Plugin
+
+The Flutter framework includes a core Platform Plugin for foundational
+platform-specific services, such as:
+
+* [Clipboard](https://docs.flutter.io/flutter/services/Clipboard-class.html):
+Utility methods for interacting with the system's clipboard.
+
+* [SystemSound](https://docs.flutter.io/flutter/services/SystemSound-class.html):
+Provides access to the library of short system specific sounds for common tasks.
+
+* [pathProvider](https://docs.flutter.io/flutter/services/PathProvider-class.html):
+Returns commonly used locations on the filesystem.
+
+See the [services library](https://docs.flutter.io/flutter/services/services-library.html)
+for full details.
+
+**Example: Using the Flutter Platform plugin to read the contents of the Clipboard**
+
+The core Flutter Platform Plugin is available through the [services
+library](https://docs.flutter.io/flutter/services/services-library.html), as
+illustrated in this example:
+
+```dart
+import 'package:flutter/services.dart';
+
+class _MyHomePageState extends State<MyHomePage> {
+ String clipboardContents = 'Unknown';
+
+ getClipboardContents() async {
+   var clipboardData = await Clipboard.getData(Clipboard.kTextPlain);
+   if (clipboardData != null) {
+     setState(() {
+       clipboardContents = clipboardData.text;
+     });
+   }
+ }
+ ...
+}
+```
+
+### Platform Plugins from the Flutter ecosystem
+
+{% include note.html content="Sharing of Platform Plugins is currently not
+supported, but Flutter is expected to support this in the Future." %}
+
+## Create a Platform Plugin {#create}
+
+Flutter uses a flexible system that allows you to call *any* platform-specific
+API whether available in Java-code on Android, or in ObjectiveC- or Swift-code
+on iOS.
+
+Flutter's platform-specific API support does not rely on fragile code
```

Comment:
Can we write this without the word "fragile" ? Some people might love code generation. Let's not make a judgement here, and just state that "does not rely on code generation".